### PR TITLE
update email template for join group requests

### LINKED ISSUE #2275: Group membership request email message: more information needed
When a user requests membership to a group, the current message only includes the first name., e.g.

Dear Christina
User Heidi has requested to join your HydroShare group (Landlab).
Click here to find information about the requesting user.
Click on the link below to accept this request.
https://www.hydroshare.org/hsapi/_internal/group_membership/4on-4438f415960b31cd6a31/77/493/?next=/
The HydroShare Team

Now that we are enforcing more information to be added, please include this in the email so the Group owner knows this is a legitimate request by adding
Last Name
Institution
Position

For example, 

Dear Christina
Heidi Cunnick, Lehigh University Graduate Student, has requested to join your HydroShare group (Landlab).

### LINKED ISSUE #3806: Enhancement to group joining functionality
The following is suggested based on discussion of groups and using them to enable app whitelists

1. When a user clicks "Join" to request group membership, a window should open "Please Justify your request" with a text box for a user to fill out.  This text should be included in the email to group owners.  
2. Optionally as one of the properties of a group there be a check box (defaulted on) indicating whether a Justification text is required or not.  If not, the behavior would be unchanged from present.

### PR DESCRIPTION
**Pull request by devincowan**
### Pull Request Checklist: 
- [X] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Navigate to Collaborate -> Groups and "ask to join" a group that your user is not a member of
2. Check debugger or Hydroshare Docker container logs: you will see that additional requested information has been added.
3. Added info includes:
    * Last name
    * Organization
    * Country
    * State
    * Email
    * Title

#### BEFORE:
before

#### AFTER:
after

Resolves #2275

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4511](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4511)

**Comment by devincowan** [comment]
I am checking in with stakeholders to see whether these proposed modifications are adequate or whether we need more.

**Comment by Castronova** [comment]
@devincowan This looks good to me!

**Comment by devincowan** [comment]
Thanks @Castronova. @pkdash this one is now ready for review, thanks!

File: hs_core/templates/email/group_membership.html [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
 {% block main %}
     {% if group_owner %}
-        <p>{% trans "Dear " %}{{ group_owner.first_name }}</p>
-        <p>{% trans "User " %}{{ user.first_name }} has requested to join your HydroShare group ({{ group.name }}).</p>
-        <p>Click<a href="{{ request.scheme }}://{{ request.get_host }}/user/{{ user.id }}"> here </a>to find information about the requesting user.</p>
+        <p>{% trans "Dear " %}{{ group_owner.first_name }}{{ group_owner.last_name }},</p>
+        <p>{% trans "User " %}{{ user.first_name }} {{ user.last_name }} 
+            (<a href="{{ request.scheme }}://{{ request.get_host }}/user/{{ user.id }}">{{ request.scheme }}://{{ request.get_host }}/user/{{ user.id }}</a>), 
+            has requested to join your HydroShare group ({{ group.name }}).</p>
+
```
Comment (pkdash) [inline hs_core/templates/email/group_membership.html:8]: Looking at the template format in issue #3806, it seems the justification text for joining the group needs to go here. Was there a decision not to implement that?

Comment (devincowan) [reply hs_core/templates/email/group_membership.html:8]: Thanks @pkdash. I thought we decided to go without it but @Castronova if you want me to add JUSTIFICATION for group request, I can do so

**Review by pkdash — APPROVED** [review]
I am approving it for now. If you decided to include justification text, I can re-approve.

**Pull request merged**